src/wolfgame.py

```diff
@@ -5716,32 +5718,59 @@ def player_stats(cli, nick, chan, rest):
     if len(params) < 2:
         reply(cli, nick, chan, db.get_player_totals(acc, hostmask), private=True)
     else:
-        role = " ".join(params[1:])
-        if role not in var.ROLE_GUIDE.keys():
-            special_keys = {"lover"}
-            evt = Event("get_role_metadata", {})
-            evt.dispatch(var, "special_keys")
-            special_keys = functools.reduce(lambda x, y: x | y, evt.data.values(), special_keys)
-            if role.lower() in var.ROLE_ALIASES:
-                matches = (var.ROLE_ALIASES[role.lower()],)
-            else:
-                matches = complete_match(role, var.ROLE_GUIDE.keys() | special_keys)
-            if not matches:
-                reply(cli, nick, chan, messages["no_such_role"].format(role))
-                return
-            if len(matches) > 1:
-                reply(cli, nick, chan, messages["ambiguous_role"].format(", ".join(matches)))
-                return
-            role = matches[0]
+        role = complete_role(cli, nick, chan, " ".join(params[1:]))
         # Attempt to find the player's stats
-        reply(cli, nick, chan, db.get_player_stats(acc, hostmask, role))
-
+        if role:
+            reply(cli, nick, chan, db.get_player_stats(acc, hostmask, role))
+        
 @cmd("mystats", "m", pm=True)
 def my_stats(cli, nick, chan, rest):
     """Get your own stats."""
     rest = rest.split()
     player_stats.func(cli, nick, chan, " ".join([nick] + rest))
 
+@cmd("rolestats", "rstats", pm=True)
```

Comment:
Please use the new command decorator rather than cmd. All of the following stuff will need to be redone to accommodate this as well since it will be based on user objects instead of nick strings.